### PR TITLE
significantly overhaul SELinux support

### PR DESCRIPTION
**Pull request by qralston**
* Update the configure script to accept the --with-selinux option to
  enable building the SELinux module.  If --with-selinux is specified,
  but the necessary development tools to build the SELinux module are
  not available, configure will fail.

* Move authlogin_duo.te from the pam_duo subdirectory to a separate
  selinux subdirectory.  Conditionally add the selinux subdirectory to
  SUBDIRS in the top-level Makefile.am, depending on whether
  --with-selinux was used.

### CODE REVIEW

File: selinux/authlogin_duo.te [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module authlogin_duo 2.0.0;
+
+## <desc>
+##  <p>
+##  Allow sshd to use the pam_duo PAM module.
+##  </p>
+## </desc>
+gen_tunable(pam_duo_permit_sshd, false)
```
Comment (xdesai) [inline selinux/authlogin_duo.te]: can we turn this to true by default?
that will help us maintain backwards compatibility

**Comment by xdesai** [comment]
Hey @qralston
Wanted to give this one more bump just to see if you were interested in splitting this into two PRs.

**Comment by qralston** [comment]
Apologies, @xdesai; my time was consumed by a different project through 2018, and I had no cycles for this.

To be clear: what you want is a change only to ```pam_duo/authlogin_duo.te```, adding the ```pam_duo_permit_sshd``` and ```pam_duo_permit_local_login``` tunables, with ```pam_duo_permit_sshd``` defaulting to ```true``` and ```pam_duo_permit_local_login``` defaulting to ```false```. Is this correct?

**Comment by xdesai** [comment]
That sounds perfect!
Also no need to apologize. We appreciate your interest in the project at whatever pace you're able to contribute.

**Comment by qralston** [comment]
@xdesai, I think what makes the most sense here is to fork just the addition of tunables out into a separate pull request. Once that pull request goes through, then I can rebase this pull request to the latest master.

**Comment by qralston** [comment]
OK, #144 contains just the addition of tunables.

**Comment by qralston** [comment]
OK, I've rebased this pull request to the latest master (ee5288d5 as of now), and updated the commit message accordingly.

@xdesai, thoughts on merging?

**Comment by qralston** [comment]
Also, FWIW, I filed support case 00295748 on 2018-10-19 in order to raise the priority of merging these SELinux enhancements. I was told that a feature request had been submitted internally, but I haven't heard anything since.

**Pull request closed without merging**